misc/python/materialize/checks/materialized_views.py

```diff
@@ -65,3 +67,83 @@ def validate(self) -> Testdrive:
            """
             )
         )
+
+
+class MaterializedViewsAssertNotNull(Check):
+    def _can_run(self, e: Executor) -> bool:
+        # CREATE ROLE not compatible with older releases
+        return self.base_version >= MzVersion.parse("0.73.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > CREATE TABLE not_null_table (x INT, y INT, z INT);
+                > INSERT INTO not_null_table VALUES (NULL, 2, 3), (4, NULL, 6), (7, 8, NULL);
+                > CREATE MATERIALIZED VIEW not_null_view1 WITH (ASSERT NOT NULL x) AS SELECT * FROM not_null_table;
+            """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW not_null_view2 WITH (ASSERT NOT NULL y) AS SELECT * FROM not_null_table;
+                > INSERT INTO not_null_table VALUES (NULL, 12, 13), (14, NULL, 16), (17, 18, NULL);
+                """,
+                """
+                > CREATE MATERIALIZED VIEW not_null_view3 WITH (ASSERT NOT NULL z) AS SELECT * FROM not_null_table;
+                > INSERT INTO not_null_table VALUES (NULL, 22, 23), (24, NULL, 26), (27, 28, NULL);
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                ! SELECT * FROM not_null_view1
```

Comment:
Can you add
* `SELECT * FROM not_null_view1 WHERE x IS NOT NULL`
* `SELECT y FROM not_null_view1`

Reply:
Sure, but what are they supposed to show? That the assert is still done?

Reply:
yeah